### PR TITLE
Getting started + JS API section

### PR DESCRIPTION
**Pull request by giann**
See also [#14](https://github.com/fengari-lua/fengari-web/pull/14)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,10 +11,30 @@
 
 🐺 φεγγάρι - The Lua VM written in JS ES6 for Node and the browser
 
+## Getting started in the browser
+
+Fengari is not released yet but you can experiment with it in the browser by following the instructions of the [fengari-web](https://github.com/fengari-lua/fengari-web) repository.
```
Comment (daurnimator) [inline README.md]: I think we need a brief intro for all the other fengari repositories.
e.g. 

This repository contains the core fengari code (which is a port of the Lua C library) which includes parser, virtual machine and base libraries.
However it is rare to use the code in this repository directly.

  - To use fengari in a web browser as easily as you might use JavaScript, see [fengari-web](https://github.com/fengari-lua/fengari-web)
  - For a clone of the `lua` command line tool, but running under node.js, see [fengari-node-cli](https://github.com/fengari-lua/fengari-node-cli)
  - [fengari-interop](https://github.com/fengari-lua/fengari-interop) is a lua library that makes interoperating with JavaScript objects simple, it is already included in fengari-web.

Comment (giann) [reply README.md]: See force pushed changes

File: README.md [hunk 2]
```diff
@@ -11,9 +11,36 @@
...
+
 ## Semantics
 
-Fengari implements Lua 5.3 semantics and will hopefully follow future Lua releases. If you find any noticeable difference between Fengari and Lua's behaviours, please [report it](https://github.com/fengari-lua/fengari/issues).
+This repository contains the core fengari code (which is a port of the Lua C library) which includes parser, virtual machine and base libraries.
```
Comment (daurnimator) [inline README.md]: This doesn't belong under `Semantics` (should be higher)

File: README.md [hunk 3]
```diff
@@ -11,9 +11,36 @@
...
+However it is rare to use this repository directly.
```
Comment (daurnimator) [inline README.md]: Need newline before list

File: README.md [hunk 4]
```diff
@@ -11,9 +11,36 @@
...
+- To use fengari in a web browser as easily as you might use JavaScript, see [fengari-web](https://github.com/fengari-lua/fengari-web)
+- [fengari-interop](https://github.com/fengari-lua/fengari-interop) is a lua library that makes interoperating with JavaScript objects simple, it is already included in fengari-web.
+- For a clone of the `lua` command line tool, but running under node.js, see [fengari-node-cli](https://github.com/fengari-lua/fengari-node-cli)
+
+To get started using fengari in the browser, you can follow [the instructions of the fengari-web repository](https://github.com/fengari-lua/fengari-web/blob/master/README.md).
```
Comment (daurnimator) [inline README.md]: Seems like duplication of first dot point

File: README.md [hunk 5]
```diff
@@ -11,9 +11,36 @@
...
+const lua      = fengari.lua;
+const lauxlib  = fengari.lauxlib;
+const lualib   = fengari.lualib;
+
+const L = lauxlib.luaL_newstate();
+
+lualib.luaL_openlibs(L);
+
+lua.lua_pushliteral(L, "hello world!");
+```
```
Comment (daurnimator) [inline README.md]: Need blank line before and after code block

File: README.md [hunk 6]
```diff
@@ -11,9 +11,36 @@
...
+This repository contains the core fengari code (which is a port of the Lua C library) which includes parser, virtual machine and base libraries.
+However it is rare to use this repository directly.
+- To use fengari in a web browser as easily as you might use JavaScript, see [fengari-web](https://github.com/fengari-lua/fengari-web)
+- [fengari-interop](https://github.com/fengari-lua/fengari-interop) is a lua library that makes interoperating with JavaScript objects simple, it is already included in fengari-web.
+- For a clone of the `lua` command line tool, but running under node.js, see [fengari-node-cli](https://github.com/fengari-lua/fengari-node-cli)
+
+To get started using fengari in the browser, you can follow [the instructions of the fengari-web repository](https://github.com/fengari-lua/fengari-web/blob/master/README.md).
+
+
+### The JS API
```
Comment (daurnimator) [inline README.md]: Also needs moving out of Semantics section.

File: README.md (hunk 2 above)
Comment (giann) [reply README.md]: oh this is a rebasing error

File: README.md [hunk 7]
```diff
@@ -11,9 +11,34 @@
...
-Fengari implements Lua 5.3 semantics and will hopefully follow future Lua releases. If you find any noticeable difference between Fengari and Lua's behaviours, please [report it](https://github.com/fengari-lua/fengari/issues).
+const L = lauxlib.luaL_newstate();
+
+lualib.luaL_openlibs(L);
+
+lua.lua_pushliteral(L, "hello world!");
+```
+
+The JS API is exactly the same as the C API so `fengari.lua` exposes the same constants and functions as `lua.h`, `fengari.lauxlib` the same as `lauxlib.h` and `fengari.lualib` the same as `lualib.h`. If you're unfamiliar with the C API, you can take a look at [the manual](http://www.lua.org/manual/5.3/manual.html#4).
+
```
Comment (daurnimator) [inline README.md]: Need 2 newlines before next higher heading

File: README.md [hunk 8]
```diff
@@ -11,9 +11,34 @@
 
 🐺 φεγγάρι - The Lua VM written in JS ES6 for Node and the browser
 
-## Semantics
+## Getting started in the browser
```
Comment (daurnimator) [inline README.md]: Need new section title

File: README.md [hunk 9]
```diff
@@ -11,9 +11,34 @@
...
+### The JS API
+
+Once you've loaded fengari, you can use the JS API:
+
+```javascript
+const lua      = fengari.lua;
+const lauxlib  = fengari.lauxlib;
+const lualib   = fengari.lualib;
 
-Fengari implements Lua 5.3 semantics and will hopefully follow future Lua releases. If you find any noticeable difference between Fengari and Lua's behaviours, please [report it](https://github.com/fengari-lua/fengari/issues).
```
Comment (daurnimator) [inline README.md]: We should keep this line.

File: README.md [hunk 10]
```diff
@@ -11,6 +11,32 @@
...
+This repository contains the core fengari code (which is a port of the Lua C library) which includes parser, virtual machine and base libraries.
+However it is rare to use this repository directly.
+
+- To use fengari in a web browser as easily as you might use JavaScript, see [fengari-web](https://github.com/fengari-lua/fengari-web)
+- [fengari-interop](https://github.com/fengari-lua/fengari-interop) is a lua library that makes interoperating with JavaScript objects simple, it is already included in fengari-web.
+- For a clone of the `lua` command line tool, but running under node.js, see [fengari-node-cli](https://github.com/fengari-lua/fengari-node-cli)
+
+### The JS API
+
+Once you've loaded fengari, you can use the JS API:
```
Comment (daurnimator) [inline README.md]: s/loaded/required ?

Comment (giann) [reply README.md]: well loaded is more generic, if you used a `<script>` you didn't do `require('fengari-web')`

**Pull request merged**